### PR TITLE
Move workshop definitions into .workshop directory

### PR DESCRIPTION
**Pull request by jonathan-conder**
# Description

Workshop definitions will now look like `.workshop/workshop.name.yaml`. The old format is no longer supported, but can be re-enabled temporarily by reverting 54e239eeb01ea756d4abe25038f255d3f2b222f6.

# Self-review quick check

 * [x] Make decisions that cost a lot to reverse explicit in the PR description.
 * [x] Avoid nested conditions.
 * [x] Delete dead code and redundant comments.
 * [x] Normalise symmetries by sticking to doing identical things identically. 
 ```
 // one way to handle errors
 if err := f(); err != nil {
    ...
 }
 
 // one way to handle multiple returns
 val, err := f()
 if err != nil {
    ...
 }
 ...
 ```
 * [x] Check that coupled code elements, files, and directories are adjacent. For example, test data is stored as close as possible to a test.
 * [x] Put variable declaration and initialisation together.
 * [x] Divide large expressions into digestable and self-explanatory ones. Use multiple variables if required.
 * [x] Put a blank line between two logically different chunks of code.

### CODE REVIEW

**Review by akcano — APPROVED** [review]
@jonathan-conder awesome work, LGTM. I'll double-check some sections this pulse, but everything runs smoothly and seems to be in order at first glance.

File: internal/overlord/workshopstate/manager.go [hunk 1]
```diff
@@ -133,9 +133,9 @@ func (w *WorkshopManager) WorkshopHealth(ws *workshop.Workshop) healthstate.Heal
 	}
 
 	// Check if the associated workshop file exists. We only check if that file
-	// exists in the project directory here; its state (e.g. if it is in sync
+	// exists in the .workshop directory here; its state (e.g. if it is in sync
 	// with the workshop instance or has any errors) is not checked.
-	path := filepath.Join(ws.Project.Path, workshop.Filename(ws.Name))
+	path := filepath.Join(ws.Project.Path, workshop.Directory, workshop.Filename(ws.Name))
```
Comment (dmitry-lyfar) [inline internal/overlord/workshopstate/manager.go]: That feels like deserving a method for the Workshop which would return its full filepath so you don't have to join in the client code.

File: internal/overlord/workshopstate/manager_test.go [hunk 2]
```diff
@@ -130,10 +134,10 @@ func (s *managerSuite) TestWorkshopHealthMissingFile(c *check.C) {
 	s.state.Lock()
 	defer s.state.Unlock()
 
-	workshop := s.launchWorkshopWithSDKs(c, "test", nil)
-	path := filepath.Join(s.project.Path, ".workshop.test.yaml")
+	testWorkshop := s.launchWorkshopWithSDKs(c, "test", nil)
+	path := filepath.Join(s.project.Path, workshop.Directory, workshop.Filename(testWorkshop.Name))
```
Comment (dmitry-lyfar) [inline internal/overlord/workshopstate/manager_test.go]: this would also be easier if the workshop had a filepath method:
```
testWorkshop.Filepath()
```

**Review by dmitry-lyfar — APPROVED** [review]
Great stuff. Thank you for updating the documentation as well. It feels so much cleaner to have workshop definitions under .workshop/.

**Review by dmitry-lyfar — APPROVED** [review]

**Comment by jonathan-conder** [comment]
I can reproduce the spread failure locally (but it was working a few hours ago...), rerunning with `-debug` now

**Pull request merged**